source/gui/__init__.py

```diff
@@ -47,6 +50,7 @@
 ### Globals
 mainFrame = None
 isInMessageBox = False
+_hasAppExited = False

```

Comment:
Is this still needed given that we're not registering our own window message?

Reply:
yes as `WM_QUIT` will exit the mainloop of wx, without calling up cleanup code, so we need to have a `safeAppExit` call after the mainloop is exited. This is done in `gui.terminate`, but I think we should only be performing `safeAppExit` once.